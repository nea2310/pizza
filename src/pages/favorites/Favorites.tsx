import ProductCard from '../../components/cards/product-card/ProductCard';
import { useAppSelector } from '../../hooks';
import { ICartItem, IPizzaDetails } from '../../interface';
import './favorites.scss';

const Favorites: React.FC = () => {
  const { userDocID, favorites } = useAppSelector((state) => state.user);
  const { pizzaItemsAll } = useAppSelector((state) => state.pizzaItems.data);
  const cart = useAppSelector((state) => state.cart.cartItems);

  /*создаем карточки купонов*/
  let pizzaItemsCards = null;
  pizzaItemsCards = pizzaItemsAll.map((pizzaItem: IPizzaDetails) => {
    let inFav = false;
    /*проверить, находится ли эта пицца в избранном*/
    if (favorites) {
      inFav = favorites.some((favItemID: string) => {
        return pizzaItem.id === favItemID;
      });
    }

    let inCart = false;
    /*проверить, находится ли эта пицца в корзине*/

    if (cart) {
      inCart = cart.some((cartItem: ICartItem) => {
        return pizzaItem.id in cartItem;
      });
    }

    if (inFav && userDocID)
      return (
        <li className="favorites__product-card" key={pizzaItem.id}>
          <ProductCard
            pizzaid={pizzaItem.id}
            isavl={pizzaItem.available}
            name={pizzaItem.name}
            image={pizzaItem.image}
            price={pizzaItem.price}
            incart={inCart}
            userdocid={userDocID}
            infav={inFav}
          />
        </li>
      );
  });

  return (
    <>
      <h1 className="favorites__main-header">Заказать пиццу</h1>
      <ul className="favorites__product-cards">{pizzaItemsCards}</ul>
    </>
  );
};

export default Favorites;
