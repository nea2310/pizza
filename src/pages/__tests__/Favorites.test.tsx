import { BrowserRouter } from 'react-router-dom';
import Favorites from '../favorites/Favorites';
import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from '../../shared/testUtils/testUtils';
import { initialStore } from '../../shared/testUtils/mockedStore';

describe('Favorites page rendering', () => {
  it('Renders Favorites page', async() => {
      const favoritesPage = renderWithProviders(<BrowserRouter><Favorites /></BrowserRouter>, {
        preloadedState: initialStore
      });

      expect(favoritesPage).toMatchSnapshot();
  });
});
