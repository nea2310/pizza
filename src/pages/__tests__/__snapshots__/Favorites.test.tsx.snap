// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page rendering Renders Favorites page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="favorites__main-header"
      >
        Заказать пиццу
      </h1>
      <ul
        class="favorites__product-cards"
      >
        <li
          class="favorites__product-card"
        >
          <article
            class="product-card"
          >
            <div
              class="product-card__favorites-button"
            >
              <button
                aria-label="добавить в избранное"
                class="favorites-button favorites-button_in-fav"
              />
            </div>
            <div
              class="product-card__content"
            >
              <h2
                class="product-card__header"
              >
                <a
                  class="product-card__link"
                  href="/product/margaritaPizzaID"
                >
                  Маргарита
                </a>
              </h2>
              <img
                alt="Маргарита"
                class="product-card__img product-card__img_inactive"
                src="https://firebasestorage.googleapis.com/margaritaPizza/"
              />
            </div>
            <div
              class="product-card__bottom"
            >
              <div
                class="product-card__info"
              >
                <span
                  class="product-card__price"
                >
                  Цена: 
                  300
                  р.
                </span>
                <span
                  class="product-card__avl"
                >
                  Нет в наличии
                </span>
              </div>
              <div
                class="product-card__order-button"
              >
                <button
                  class="order-button order-button_type_order"
                  disabled=""
                >
                  Заказать
                </button>
              </div>
            </div>
          </article>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1
      class="favorites__main-header"
    >
      Заказать пиццу
    </h1>
    <ul
      class="favorites__product-cards"
    >
      <li
        class="favorites__product-card"
      >
        <article
          class="product-card"
        >
          <div
            class="product-card__favorites-button"
          >
            <button
              aria-label="добавить в избранное"
              class="favorites-button favorites-button_in-fav"
            />
          </div>
          <div
            class="product-card__content"
          >
            <h2
              class="product-card__header"
            >
              <a
                class="product-card__link"
                href="/product/margaritaPizzaID"
              >
                Маргарита
              </a>
            </h2>
            <img
              alt="Маргарита"
              class="product-card__img product-card__img_inactive"
              src="https://firebasestorage.googleapis.com/margaritaPizza/"
            />
          </div>
          <div
            class="product-card__bottom"
          >
            <div
              class="product-card__info"
            >
              <span
                class="product-card__price"
              >
                Цена: 
                300
                р.
              </span>
              <span
                class="product-card__avl"
              >
                Нет в наличии
              </span>
            </div>
            <div
              class="product-card__order-button"
            >
              <button
                class="order-button order-button_type_order"
                disabled=""
              >
                Заказать
              </button>
            </div>
          </div>
        </article>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
